src/main/webapp/creatives/maps/simid_map_creative.js

```diff
@@ -84,6 +83,53 @@ export default class SimidMapCreative extends BaseSimidCreative {
    * @private 
   */
   grantLocationAccess_() {
-    //ToDo(kristenmason@): implement map
+    //ToDo(kristenmason@): implement the Google Maps request access functionality
+    this.simidProtocol.sendMessage(CreativeMessage.REQUEST_PAUSE).then(() => {
+      findNearest.classList.add("hidden");
+    }).catch(() => {
+      this.simidProtocol.reject(eventData, {errorCode: CreativeErrorCode.PAUSE_NOT_HONORED, 
+      message: "Request to pause ad not honored"
+      });
+    });
+    this.createButtonsMapState_();
+  }
+
+  /**
+   * Creates the Skip To Content and Return To Ad buttons once the user
+   *   grants permission to access their location and the map appears.
+   * @private 
+  */
+  createButtonsMapState_() {
+    const returnToAdButton = document.createElement("button");
+    returnToAdButton.textContent = "Return To Ad";
+    returnToAdButton.id = "returnToAd";
+    returnToAdButton.onclick = () => this.playAd_(returnToAdButton); 
+
+    const skipAdButton = document.createElement("button");
+    skipAdButton.textContent = "Skip Ad";
+    skipAdButton.id = "skipToContent";
+    skipAdButton.onclick = () => this.playContent_();
+
+    const adContainer = document.getElementById('adContainer');
+    adContainer.appendChild(returnToAdButton);
+    adContainer.appendChild(skipAdButton);
+  }
+
+  /**
+   * Continues to play the ad if user clicks on Return To Ad button.
+   * @private 
+  */
+  playAd_(returnToAdButton) {
```

Comment:
Just to make sure im clear on this logic -- looks like the skip button only shows up if the user grants their location access, but then if they go back to the ad, the skip button stays visible? Is that all intentional?

Reply:
Yes that is intentional- when the map appears both buttons return to ad and skip are displayed. If the user clicks return to ad, then I hide that button because it would look weird to have that stay put if they already returned to the ad. However, just because a user returns to the ad, doesn't mean they should now be stripped of the opportunity to skip it, so I continue displaying the skip button for the duration of the ad